reports/serializers.py

```diff
@@ -1,8 +1,51 @@
+from typing import List
+
 from reports.models import EnrolmentReport
 from rest_framework import serializers
 
 
+class Size2ArrayFieldBase(serializers.Field):
+    def __init__(self, first_value_name: str = "id", second_value_name: str = "label"):
+        self.first_value_name = first_value_name
+        self.second_value_name = second_value_name
+        super().__init__()
+
+    def _array_to_dict(self, entry: List[str]):
+        if len(entry) == 2:
+            return {self.first_value_name: entry[0], self.second_value_name: entry[1]}
+        return None
+
+    def _dict_to_array(self, entry: dict):
+        if entry:
+            return [entry[self.first_value_name], entry[self.second_value_name]]
+        return None
+
+    def to_representation(self, value):
+        return [self._array_to_dict(entry) for entry in value] if value else None
+
+    def to_internal_value(self, data):
+        return [self._dict_to_array(entry) for entry in data] if data else None
+
+
+class StudyGroupStudyLevelsField(Size2ArrayFieldBase):
+    pass
+
+
+class OccurrenceLanguagesField(Size2ArrayFieldBase):
+    def __init__(self):
+        super().__init__("id", "name")
+
+
+class EventKeywordsField(Size2ArrayFieldBase):
+    def __init__(self):
+        super().__init__("id", "name")
```

Comment:
Not sure why these subclasses are needed, wouldn't it be easier to just use `Size2ArrayField` in serializers like `keywords = Size2ArrayField(second_value_name="name")`.